### DISCUSSION TITLE
Gifting to a player will always arrive on the next day even if the receiving player has a turn before that

### DISCUSSION THREAD

**Discussion by zenseii** [ihhub/fheroes2#7579] [General]
### Preliminary checks

- [X] I've checked that there aren't [**other open issues**](https://github.com/ihhub/fheroes2/issues?q=is%3Aissue) on the same topic.

### Describe the problem requiring a solution

Playing hotseat mode and controlling all players, if you gift resources then they will not arrive within the same day and so the firstcoming turn of that player, but the next turn of the next day of the receiving player(s). This could end up meaning they get it on their second turn after the gifting is done. If the giving player comes after the receiving player in the turn order, then the receiving player will get the resources on the firstcoming turn of that player which would be the next day after the resources are sent.

### Describe the possible solution

The gift should arrive on the first turn of the receiving player after the gift is sent.

**Comment by oleg-derevenetz** [comment]
Hi @zenseii gifted resources will arrive on the next DAY. Please don't mix DAY and TURN :) While the red player has a TURN after the blue player, it is still the same DAY, so he will need to wait for the next DAY for gifted resources to arrive.

**Comment by zenseii** [comment]
Right, thanks! I corrected it.

**Comment by oleg-derevenetz** [comment]
@zenseii I understand the logic of those who developed the gift mechanics - if you make it so that resources come just on the next turn, then sometimes they will come on the same day, and sometimes on the next day (depending on the sequence of the giving player's and the receiving player's turn), because time cannot go back. Now everything happens simply and logically - moving resources from one player to another always takes one day, and receiving player will always get them in his turn on the next day. I propose to leave this logic as it is.

**Comment by zenseii** [comment]
> @zenseii I understand the logic of those who developed the gift mechanics - if you make it so that resources come just on the next turn, then sometimes they will come on the same day, and sometimes on the next day (depending on the sequence of the giving player's and the receiving player's turn), because time cannot go back. Now everything happens simply and logically - moving resources from one player to another always takes one day, and receiving player will always get them in his turn on the next day. I propose to leave this logic as it is.

@oleg-derevenetz I agree that there is a logic to it. I still believe the gift should come on the immediate next turn of that receiving player, whether that is within the same day or the next day. This is similar to how if you attack a player in your turn before they have had theirs, your attack won't happen the next day just because they haven't had their day yet. This is sort of a conundrum of the gameplay in Heroes in general, but it should try to be consistent in its approach to this. Not to mention that having to plan in such a way ahead of giving/receiving a gift often makes it useless because unforseen things happen.

This way of gifting the next turn is also how it works in later Heroes games so that's what players will be used to, which is something worth considering.

**Comment by Branikolog** [comment]
Hello, @zenseii .
I agree with @oleg-derevenetz . We should keep clear and obvious logic for this "gift" mechanics.
While playing with allies there could be advantage for the players, that start earlier, which is unfair.

**Comment by zenseii** [comment]
> Hello, @zenseii . I agree with @oleg-derevenetz . We should keep clear and obvious logic for this "gift" mechanics. While playing with allies there could be advantage for the players, that start earlier, which is unfair.

Yes, this is true. I already thought about there being an advantage if you are first in the turn order. However, this is also the case for many other things regarding the game and the turn order. There are also disadvantages to being the first player.

Personally, I don't mind it being either way. Balancing aspects like these can only be done after lots of playtesting, and for this aspect such a thing will only be possible after online multiplayer has been implemented, because the amount of players playing in hotseat and against each other would be too few.

**Comment by ihhub** [comment]
Shall we just state that resource gifting takes 1 day? Something like "Resource Guild needs 1 day to process your request".

**Comment by zenseii** [comment]
> Shall we just state that resource gifting takes 1 day? Something like "Resource Guild needs 1 day to process your request".

Sure. Where would we put this?

**Comment by ihhub** [comment]
I think in the same window where we select recipients. We might need to expand the window.

**Comment by zenseii** [comment]
I'm converting this issue to a discussion since there is no clear agreement on what to change if anything.
